prediction_market_agent/agents/microchain_agent/nft_treasury_game/app_nft_treasury_game.py

```diff
@@ -178,12 +181,39 @@ def customized_chat_message(
             st.markdown(parsed_function_output_body)
 
 
-@st.fragment(run_every=timedelta(seconds=5))
 def show_function_calls_part(nft_agent: type[DeployableAgentNFTGameAbstract]) -> None:
     st.markdown(f"""### Agent's actions""")
 
+    messages_per_page = 50
+    if "page_number" not in st.session_state:
+        st.session_state.page_number = 0
+
+    col1, col2, col3 = st.columns(3)
+    with col1:
+        if st.button("Previous page", disabled=st.session_state.page_number == 0):
+            st.session_state.page_number -= 1
+    with col2:
+        if st.button("Next page"):
+            st.session_state.page_number += 1
```

Comment:
I would expect this button to be disabled if last page was reached - is that the case?

Reply:
It does now 😄